### ISSUE TITLE
Group 07 - chembox

### ISSUE THREAD

**Issue by Suraporn** [UBC-MDS/software-review-2023#9]
Submitting Author: NatePuangpanbut (@Suraporn)
All current maintainers: Wilfred Hass (@WilfHass), Vikram Grewal (@xFiveRivers), Luke Yang (@lukeyf)
Package Name: chembox
One-Line Description of Package: Molecular information calculator based on empirical formulas of chemicals in raw text.
Repository Link: https://github.com/UBC-MDS/chembox 
Version submitted: v0.3.2  
Editor: Wilfred Hass (@WilfHass), Vikram Grewal (@xFiveRivers), Luke Yang (@lukeyf), NatePuangpanbut (@Suraporn) 
Reviewer 1: Mehwish
Reviewer 2: Robin Dhillon
Reviewer 3: Flora Ouedraogo
Reviewer 4: Yukun Zhang 
Archive: TBD  
Version accepted: TBD 
Date accepted (month/day/year): TBD

---

## Description

`chembox` is a package for molecular information calculator based on empirical formulas of chemicals in raw text. It is designed to intelligently process text input containing the chemical formula and provide associated information on the inquired molecule. It is able to provide the molar mass, check a formula's validity, and provide a balanced combustion equation if the input is combustible. This tool can be utilized for various educational and research purposes for simple and fast information retrieval.

## Scope 
- Please indicate which [category or categories][PackageCategories] this package falls under:
	- [ ] Data retrieval
	- [x] Data extraction
	- [ ] Data munging
	- [ ] Data deposition
	- [ ] Reproducibility
	- [ ] Geospatial
	- [ ] Education
	- [ ] Data visualization*

> *Please fill out a pre-submission inquiry before submitting a data visualization package. For more info, see [notes on categories][NotesOnCategories] of our guidebook.*

- **For all submissions**, explain how the and why the package falls under the categories you indicated above. In your explanation, please address the following points (briefly, 1-2 sentences for each):  

  - Who is the target audience and what are scientific applications of this package?  
     The target audience is anyone who is interested in extracting chemical properties from a chemical formula. It can high school students, undergraduate students, or any chemistry enthusiastic people who want to learn about chemical formula properties.

  - Are there other Python packages that accomplish the same thing? If so, how does yours differ?
     A similar package [chemsolve](https://github.com/amogh7joshi/chemsolve) is available online that employs a similar string-parsing design and molar mass calculation. Yet, in `chembox`, we include a dataframe of the molecule properties to give the user enhanced functionality and insights. In addition, we include methods for automated combustion equation generation and balancing.

  - If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or `@tag` the editor you contacted:

## Technical checks

For details about the pyOpenSci packaging requirements, see our [packaging guide][PackagingGuide]. Confirm each of the following by checking the box.  This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] has an [OSI approved license][OsiApprovedLicense].
- [x] contains a README with instructions for installing the development version. 
- [x] includes documentation with examples for all functions.
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a test suite.
- [x] has continuous integration, such as Travis CI, AppVeyor, CircleCI, and/or others.

## Publication options

- [ ] Do you wish to automatically submit to the [Journal of Open Source Software][JournalOfOpenSourceSoftware]? If so:

<details>
 <summary>JOSS Checks</summary>  

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements][JossSubmissionRequirements]. Be aware that completing the pyOpenSci review process **does not** guarantee acceptance to JOSS. Be sure to read their submission requirements (linked above) if you are interested in submitting to JOSS.
- [ ] The package is not a "minor utility" as defined by JOSS's [submission requirements][JossSubmissionRequirements]: "Minor ‘utility’ packages, including ‘thin’ API clients, are not acceptable." pyOpenSci welcomes these packages under "Data Retrieval", but JOSS has slightly different criteria.
- [ ] The package contains a `paper.md` matching [JOSS's requirements][JossPaperRequirements] with a high-level description in the package root or in `inst/`.
- [ ] The package is deposited in a long-term repository with the DOI: 

*Note: Do not submit your package separately to JOSS*
  
</details>

## Are you OK with Reviewers Submitting Issues and/or pull requests to your Repo Directly?
This option will allow reviewers to open smaller issues that can then be linked to PR's rather than submitting a more dense text based review. It will also allow you to demonstrate addressing the issue via PR links.

- [x] Yes I am OK with reviewers submitting requested changes as issues to my repo. Reviewers will then link to the issues in their submitted review.

## Code of conduct

- [x] I agree to abide by [pyOpenSci's Code of Conduct][PyOpenSciCodeOfConduct] during the review process and in maintaining my package should it be accepted.

## Please fill out our survey
- [ ] [Last but not least please fill out our pre-review survey](https://forms.gle/F9mou7S3jhe8DMJ16). This helps us track
submission and improve our peer review process. We will also ask our reviewers 
and editors to fill this out.

**P.S.** *Have feedback/comments about our review process? Leave a comment [here][Comments]

## Editor and Review Templates

The [editor template can be found here][Editor Template].

The [review template can be found here][Review Template].

[PackagingGuide]: https://www.pyopensci.org/contributing-guide/authoring/index.html#packaging-guide

[PackageCategories]: https://www.pyopensci.org/contributing-guide/open-source-software-peer-review/aims-and-scope.html?highlight=data#package-categories

[NotesOnCategories]: https://www.pyopensci.org/contributing-guide/open-source-software-peer-review/aims-and-scope.html?highlight=data#notes-on-categories

[JournalOfOpenSourceSoftware]: http://joss.theoj.org/

[JossSubmissionRequirements]: https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements

[JossPaperRequirements]: https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain

[PyOpenSciCodeOfConduct]: https://www.pyopensci.org/contributing-guide/open-source-software-peer-review/code-of-conduct.html?highlight=code%20conduct

[OsiApprovedLicense]: https://opensource.org/licenses

[Editor Template]: https://www.pyopensci.org/peer-review-guide/software-peer-review-guide/editors-guide.html#respond-to-the-submitter-in-the-github-issue

[Review Template]: https://www.pyopensci.org/peer-review-guide/software-peer-review-guide/reviewer-guide.html#peer-review-template

[Comments]: https://github.com/pyOpenSci/governance/issues/8

**Comment by robindhillon1** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [x] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [ ] Python versions supported,
    - [ ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [x] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [ ] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [x] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 2 hours

---

#### Review Comments

- First of all, you guys had a code coverage of 98% which is simply awesome. Good job!
- The package installed smoothly and the examples in the usage section worked well.
    - ![image](https://user-images.githubusercontent.com/86622951/216715934-8bf1ffc2-3900-45b3-98bd-b57e1a12c1fd.png)
- I do have 2 small suggestions regarding the package installation and usage section, however:
    - Include a small example in the Usage section in the `README` itself and then segue into the ReadTheDocs example, as I think that is a bit more clear and works with the flow as the reader reads. Initially, I thought there were no examples until I found the link to the example. 
    -  For someone with very little or no experience, they might assume that they only have to install the package via `pip` and that the examples in the usage will work. However, we know that only works if the repository is cloned and the example commands are run in its root directory. A layman might not know this, so an explicit instruction such as: "Clone the repo and then run the following commands in its root directory", for example, would work better, I believe.
- There doesn't seem to be a link that redirects to the main page of ReadTheDocs. This should probably be added to the `README` somewhere. Here's the link: https://chembox.readthedocs.io/en/latest/index.html
- The hyperlink in the `Dataset` section in the `README` redirects to data that doesn't exist. Here's the link which gives me a 404 error: https://github.com/UBC-MDS/chembox/tree/main/src/chembox/data. This is an easy fix.
- I especially like the fact that you have one file containing all the tests; the testing is very thorough which is impressive. All the functions exist in one script as well, which is efficient. The documentation of each function is clean and easy to understand. Very pleasing to look at as well.
- Ultimately, this is a very well-written and thorough package; apart from the small aforementioned changes/fixes, a near-perfect package. Amazing work.

**Comment by florawendy19** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.


#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    


Estimated hours spent reviewing:
1 hour

---

#### Review Comments

1. I was able to install the python chembox package  without any issue locally. I was able to call `get_elements` ,` is_valid`, `get_molec_props`, `get_combustion_equation` and it gave the appropriate results as indicated in the readme file. This  is a really great idea for a package as it will make it easier for students to easily find things such as  molar mass and balanced  combustion.
2. It is really great that you mentioned in the README packages that already exist and how yours is different. it shows that you have done some research  prior developing your package in order to come up with a more innovative way of addressing the need that you have identified. 

I do not have anything technical to add as the package is working well and the functions give the correct results. I just have few remarks about your repo which are the following:

1.  I wish there the different functions were in different python files in the src folder. everything is in block and it is easy to get lost in these several lines of code. 
2. It would have been good to also have the output of one or two functions on the README to give an idea of how the output looks like visually. 
3. I would have been also good to add some citation for you data. Does the data get updated regularly or is it always static, these are the question I have as go through the . If the data is static, where can I find it?

I enjoyed  reviewing your package even though I do not like anything related to Chemistry!  You did a really great job with this package and I wish you all the best in your software Development journey !

**Comment by yukunzGIT** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [ ] Badges for:
    - [ ] Continuous integration and test coverage,
    - [ ] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [x] Python versions supported,
    - [x] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [ ] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [x] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [x] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1 hour

---

#### Review Comments

A 98% code coverage is great. Good job, group 7. 
I also have some suggestions regarding your Python package:

1. Maybe you could save your results data frame in a csv file for future usage.
2. For get_combustion_equation function, what if the input molecule is not combustable? Maybe try to return an error message if the user input molecule is not combustable. 
3. Miss the doc description for the test function for unit tests functions of three functions. Also, maybe it's better to split the test scripts into 4 different parts for each of the four functions. 
4. Before we do the pip install, try to add "clone the repo" with the repo link in the README.md for package installation. 
5. The Read the docs link is not added in the README. Note it's good that you guys add the link to the original dataset.

In summary, I really enjoy playing around with your chem package and find it could be really helpful for beginners to learn Chemistry. Great jobs, group 7. Wish you all the best!

**Comment by MNBhat** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below. Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README.
- [x] **Installation instructions:** for the development version of the package and any non-standard dependencies in README.
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally.
- [x] **Function Documentation:** for all user-facing functions.
- [x] **Examples** for all user-facing functions.
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING.
- [x] **Metadata** including author(s), author e-mail(s), a url, and any other relevant metadata e.g., in a `pyproject.toml` file or elsewhere.

Readme file  requirements
The package meets the readme requirements below:

- [x] Package has a README.md file in the root directory.

The README should include, from top to bottom:

- [x] The package name
- [ ] Badges for:
    - [x] Continuous integration and test coverage,
    - [x] Docs building (if you have a documentation website),
    - [ ] A [repostatus.org](https://www.repostatus.org/) badge,
    - [ ] Python versions supported,
    - [ ] Current package version (on PyPI / Conda).

*NOTE: If the README has many more badges, you might want to consider using a table for badges: [see this example](https://github.com/ropensci/drake). Such a table should be more wide than high. (Note that the a badge for pyOpenSci peer-review will be provided upon acceptance.)*

- [x] Short description of package goals.
- [x] Package installation instructions
- [x] Any additional setup required to use the package (authentication tokens, etc.)
- [ ] Descriptive links to all vignettes. If the package is small, there may only be a need for one vignette which could be placed in the README.md file.
    - [x] Brief demonstration of package usage (as it makes sense - links to vignettes could also suffice here if package description is clear)
- [x] Link to your documentation website.
- [x] If applicable, how the package compares to other similar packages and/or how it relates to other packages in the scientific ecosystem.
- [x] Citation information

#### Usability
Reviewers are encouraged to submit suggestions (or pull requests) that will improve the usability of the package as a whole.
Package structure should follow general community best-practices. In general please consider whether:

- [x] Package documentation is clear and easy to find and use.
- [x] The need for the package is clear
- [x] All functions have documentation and associated examples for use
- [x] The package is easy to install


#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [ ] **Automated tests:** Tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Continuous Integration:** Has continuous integration setup (We suggest using Github actions but any CI platform is acceptable for review)
- [x] **Packaging guidelines**: The package conforms to the pyOpenSci [packaging guidelines](https://www.pyopensci.org/python-package-guide).
    A few notable highlights to look at:
    - [x] Package supports modern versions of Python and not [End of life versions](https://endoflife.date/python).
    - [x] Code format is standard throughout package and follows PEP 8 guidelines (CI tests for linting pass)

#### For packages also submitting to JOSS

- [ ] The package has an **obvious research application** according to JOSS's definition in their [submission requirements](http://joss.theoj.org/about#submission_requirements).

*Note:* Be sure to check this carefully, as JOSS's submission requirements and scope differ from pyOpenSci's in terms of what types of packages are accepted.

The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:

- [ ] **A short summary** describing the high-level functionality of the software
- [ ] **Authors:** A list of authors with their affiliations
- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
- [ ] **References:** With DOIs for all those that have one (e.g. papers, datasets, software).

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 1.5 Hours

---

#### Review Comments
1. The functions and test functions are well documented, and the documentation of test cases is clear about what all scenarios are being checked in the test. Great work!
2. All the details on Read.me are clear, and direct toward appropriate links about where to find the information needed. 
3. The function get_molec_props() works nicely and is super clear.  I would suggest removing extra commas  ShellConfig output as it seems that there must be something missing. A clear output eg 2,8 will be much better. 
4. For some functions like get_elements() it would be great if there is additional functionality for checking inputs in various formats like get_elements('5(C2H4)'),  get_elements('(C2H4)') which right now are throwing an index error. 
5.  Some functions are not checking if the input type is string or not. It would be nice to have this checked. 
6.  Looking at this amazing package makes me think it would be awesome to have a function in the package that takes in the unbalanced equation and returns a balanced one.  
The package is well-written and well documented with amazing code coverage of 98%.  I enjoyed testing this package and loved get_molec_props() function that gives that molecular information so clearly (which was a nightmare for me in my high school).  Well done!
